src/Items/ItemHandler.cpp

```diff
@@ -251,7 +262,24 @@ cItemHandler * cItemHandler::CreateItemHandler(int a_ItemType)
 		case E_ITEM_SPIDER_EYE:
 		case E_ITEM_STEAK:
 		{
-			return new cItemFoodHandler(a_ItemType);
+			handler.push_back(cpp14::make_unique<cItemFoodHandler>(a_ItemType));
+			break;
+		}
+
+		case E_ITEM_COOKED_FISH:
+		{
+			handler.push_back(cpp14::make_unique<cItemFoodHandler>(a_ItemType, E_META_COOKED_FISH_FISH));
+			handler.push_back(cpp14::make_unique<cItemFoodHandler>(a_ItemType, E_META_COOKED_FISH_SALMON));
+			break;
+		}
+
+		case E_ITEM_RAW_FISH:
+		{
+			handler.push_back(cpp14::make_unique<cItemFoodHandler>(a_ItemType, E_META_RAW_FISH_FISH));
+			handler.push_back(cpp14::make_unique<cItemFoodHandler>(a_ItemType, E_META_RAW_FISH_SALMON));
+			handler.push_back(cpp14::make_unique<cItemFoodHandler>(a_ItemType, E_META_RAW_FISH_CLOWNFISH));
+			handler.push_back(cpp14::make_unique<cItemFoodHandler>(a_ItemType, E_META_RAW_FISH_PUFFERFISH));
```

Comment:
I was thinking like cItemCookedFish, ItemCookedSalmon, ItemRawFish, etc.

Could you just do it with meta as an argument though? The de-facto method used everywhere is for the handlers not to store anything but block/item type (i.e. see `cBlockHandler::GetMapBaseColour(NIBBLETYPE a_Meta)`) and really, `Item.GetHandler().EatFood(Item.m_Damage);` is much better than all those `push_back`s and dealing unnecessarily with vector overhead.

If you want, have separate handlers for fish and cooked fish, as tycho has suggested, but I would recommend the above approach to metadata.